teamcity: failed tests on master: testrace/TestTimeSeriesMaintenanceQueueServer, test/TestTimeSeriesMaintenanceQueueServer

The following tests appear to have failed:

[#575924](https://teamcity.cockroachdb.com/viewLog.html?buildId=575924):


```
--- FAIL: testrace/TestTimeSeriesMaintenanceQueueServer (1.620s)
ts_maintenance_queue_test.go:316: got datapoints [{1510698125000000000 100} {1515882125000000000 200}], expected [{1510698125000000000 100} {1515882125000000000 200} {1521066125000000000 300}], diff: [[]tspb.TimeSeriesDatapoint[2] != []tspb.TimeSeriesDatapoint[3]]
------- Stdout: -------
W180314 22:21:58.992608 137882 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180314 22:21:59.016582 137882 server/server.go:721  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180314 22:21:59.020469 137882 server/config.go:538  [n?] 1 storage engine initialized
I180314 22:21:59.020600 137882 server/config.go:541  [n?] RocksDB cache size: 128 MiB
I180314 22:21:59.020635 137882 server/config.go:541  [n?] store 0: in-memory, size 0 B
I180314 22:21:59.062901 137882 server/node.go:374  [n?] **** cluster 19acc337-360e-44f2-aa50-9123e7740c3a has been created
I180314 22:21:59.063088 137882 server/server.go:1110  [n?] **** add additional nodes by specifying --join=127.0.0.1:34011
I180314 22:21:59.077829 137882 storage/store.go:1394  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180314 22:21:59.078966 137882 server/node.go:504  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=4.9 KiB), ranges=1, leases=0, writes=0.00, bytesPerReplica={p10=5043.00 p25=5043.00 p50=5043.00 p75=5043.00 p90=5043.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00}
I180314 22:21:59.079456 137882 server/node.go:352  [n1] node ID 1 initialized
I180314 22:21:59.079895 137882 gossip/gossip.go:332  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:34011" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:1 > 
I180314 22:21:59.082284 137882 storage/stores.go:331  [n1] read 0 node addresses from persistent storage
I180314 22:21:59.083728 137882 server/node.go:645  [n1] connecting to gossip network to verify cluster ID...
I180314 22:21:59.084046 137882 server/node.go:670  [n1] node connected via gossip and verified as part of cluster "19acc337-360e-44f2-aa50-9123e7740c3a"
I180314 22:21:59.084482 137882 server/node.go:446  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180314 22:21:59.086743 137882 server/server.go:1228  [n1] starting https server at 127.0.0.1:41085
I180314 22:21:59.086876 137882 server/server.go:1229  [n1] starting grpc/postgres server at 127.0.0.1:34011
I180314 22:21:59.086942 137882 server/server.go:1230  [n1] advertising CockroachDB node at 127.0.0.1:34011
W180314 22:21:59.087246 137882 sql/jobs/registry.go:233  [n1] unable to get node liveness: node not in the liveness table
I180314 22:21:59.109870 138119 storage/replica_command.go:851  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I180314 22:21:59.167683 137755 storage/replica_command.go:851  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
I180314 22:21:59.223493 137882 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:node}
I180314 22:21:59.317181 138123 storage/replica_command.go:851  [split,n1,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I180314 22:21:59.389111 137873 storage/replica_command.go:851  [split,n1,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
I180314 22:21:59.452073 138096 storage/replica_command.go:851  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
W180314 22:21:59.464560 138163 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "sql txn implicit" id=d34df7ed key=/Table/SystemConfigSpan/Start rw=true pri=0.01403173 iso=SERIALIZABLE stat=PENDING epo=0 ts=1521066119.362462836,0 orig=1521066119.362462836,0 max=1521066119.362462836,0 wto=false rop=false seq=14
I180314 22:21:59.479581 137882 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:$1 User:node}
I180314 22:21:59.517332 138125 storage/replica_command.go:851  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I180314 22:21:59.563482 137882 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:node}
I180314 22:21:59.587381 138179 storage/replica_command.go:851  [split,n1,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I180314 22:21:59.650342 138169 storage/replica_command.go:851  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I180314 22:21:59.707337 138172 storage/replica_command.go:851  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180314 22:21:59.761255 138212 storage/replica_command.go:851  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180314 22:21:59.809888 138184 storage/replica_command.go:851  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180314 22:21:59.821416 137882 sql/event_log.go:121  [n1] Event: "alter_table", target: 4, info: {TableName:system.public.users Statement:ALTER TABLE system.public.users ADD COLUMN IF NOT EXISTS "isRole" BOOL NOT NULL DEFAULT false User:node MutationID:1 CascadeDroppedViews:[]}
I180314 22:21:59.864610 137882 sql/lease.go:352  [n1] publish: descID=4 (users) version=2 mtime=2018-03-14 22:21:59.8631306 +0000 UTC
I180314 22:21:59.873162 138227 storage/replica_command.go:851  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180314 22:21:59.930378 138242 storage/replica_command.go:851  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180314 22:21:59.957192 137882 sql/lease.go:352  [n1] publish: descID=4 (users) version=3 mtime=2018-03-14 22:21:59.955466602 +0000 UTC
I180314 22:21:59.981232 138258 storage/replica_command.go:851  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I180314 22:22:00.017782 137882 sql/backfill.go:134  [n1] Running backfill for "users", v=3, m=1
I180314 22:22:00.036404 138234 storage/replica_command.go:851  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I180314 22:22:00.088387 138017 storage/replica_command.go:851  [split,n1,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I180314 22:22:00.136863 138278 storage/replica_command.go:851  [split,n1,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I180314 22:22:00.144746 137882 sql/lease.go:352  [n1] publish: descID=4 (users) version=4 mtime=2018-03-14 22:22:00.143634751 +0000 UTC
I180314 22:22:00.191113 138279 storage/replica_command.go:851  [split,n1,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180314 22:22:00.217736 137882 sql/event_log.go:121  [n1] Event: "finish_schema_change", target: 4, info: {MutationID:1}
I180314 22:22:00.240157 137882 sql/lease.go:275  publish (count leases): descID=4 name=users version=3 count=1
I180314 22:22:00.252094 138312 sql/lease.go:803  new lease: 4("users") ver=4:1521066480.691648094,0, refcount=0
I180314 22:22:00.259745 138315 storage/replica_command.go:851  [split,n1,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
I180314 22:22:00.479759 137882 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:gen_random_uuid()::STRING User:node}
I180314 22:22:00.489283 137882 server/server.go:1308  [n1] done ensuring all necessary migrations have run
I180314 22:22:00.489533 137882 server/server.go:1311  [n1] serving sql connections
I180314 22:22:00.503725 137882 storage/replica_command.go:851  [n1,s1,r5/1:/System/ts{d-e}] initiating a split of this range at key /System/tsd/test.metric/source1/10s/2017-12-14T22:00:00Z [r21]
I180314 22:22:00.511433 138285 sql/event_log.go:121  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:34011} Attrs: Locality: ServerVersion:2.0-1} ClusterID:19acc337-360e-44f2-aa50-9123e7740c3a StartedAt:1521066119084129017 LastUp:1521066119084129017}
I180314 22:22:00.557073 137882 util/stop/stopper.go:473  quiescing; tasks left:
1      storage.intentResolver: processing intents
1      [async] transport racer
I180314 22:22:00.557181 138118 kv/transport_race.go:67  transport race promotion: ran 24 iterations on up to 792 requests
I180314 22:22:00.557782 137882 util/stop/stopper.go:473  quiescing; tasks left:
1      storage.intentResolver: processing intents
W180314 22:22:00.557464 138359 storage/intent_resolver.go:388  [n1,s1] failed to cleanup transaction intents: could not GC completed transaction anchored at /Local/Range/System/tsd/RangeDescriptor: node unavailable; try another peer
--- FAIL: test/TestTimeSeriesMaintenanceQueueServer (0.170s)
ts_maintenance_queue_test.go:316: got datapoints [{1510697155000000000 100} {1515881155000000000 200}], expected [{1510697155000000000 100} {1515881155000000000 200} {1521065155000000000 300}], diff: [[]tspb.TimeSeriesDatapoint[2] != []tspb.TimeSeriesDatapoint[3]]
------- Stdout: -------
W180314 22:05:53.797250 148049 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180314 22:05:53.800763 148049 server/server.go:721  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180314 22:05:53.802697 148049 server/config.go:538  [n?] 1 storage engine initialized
I180314 22:05:53.802767 148049 server/config.go:541  [n?] RocksDB cache size: 128 MiB
I180314 22:05:53.802778 148049 server/config.go:541  [n?] store 0: in-memory, size 0 B
I180314 22:05:53.806893 148049 server/node.go:374  [n?] **** cluster b973e615-8683-4de6-a792-05f17a258c47 has been created
I180314 22:05:53.806928 148049 server/server.go:1110  [n?] **** add additional nodes by specifying --join=127.0.0.1:45607
I180314 22:05:53.807768 148049 storage/store.go:1394  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180314 22:05:53.807888 148049 server/node.go:504  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=4.9 KiB), ranges=1, leases=0, writes=0.00, bytesPerReplica={p10=5043.00 p25=5043.00 p50=5043.00 p75=5043.00 p90=5043.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00}
I180314 22:05:53.807925 148049 server/node.go:352  [n1] node ID 1 initialized
I180314 22:05:53.807979 148049 gossip/gossip.go:332  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:45607" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:1 > 
I180314 22:05:53.808067 148049 storage/stores.go:331  [n1] read 0 node addresses from persistent storage
I180314 22:05:53.808198 148049 server/node.go:645  [n1] connecting to gossip network to verify cluster ID...
I180314 22:05:53.808219 148049 server/node.go:670  [n1] node connected via gossip and verified as part of cluster "b973e615-8683-4de6-a792-05f17a258c47"
I180314 22:05:53.808235 148049 server/node.go:446  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180314 22:05:53.809176 148049 server/server.go:1228  [n1] starting https server at 127.0.0.1:37913
I180314 22:05:53.809204 148049 server/server.go:1229  [n1] starting grpc/postgres server at 127.0.0.1:45607
I180314 22:05:53.809214 148049 server/server.go:1230  [n1] advertising CockroachDB node at 127.0.0.1:45607
W180314 22:05:53.809234 148049 sql/jobs/registry.go:233  [n1] unable to get node liveness: node not in the liveness table
I180314 22:05:53.813344 147875 storage/replica_command.go:851  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I180314 22:05:53.817822 148107 storage/replica_command.go:851  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
I180314 22:05:53.822739 148289 storage/replica_command.go:851  [split,n1,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I180314 22:05:53.830576 148338 storage/replica_command.go:851  [split,n1,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
I180314 22:05:53.835926 147873 storage/replica_command.go:851  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
I180314 22:05:53.842976 148341 storage/replica_command.go:851  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I180314 22:05:53.848675 148354 storage/replica_command.go:851  [split,n1,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I180314 22:05:53.853805 148348 storage/replica_command.go:851  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I180314 22:05:53.858662 148350 storage/replica_command.go:851  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180314 22:05:53.862924 148049 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:node}
I180314 22:05:53.864208 148402 storage/replica_command.go:851  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180314 22:05:53.869645 148311 storage/replica_command.go:851  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180314 22:05:53.870429 148049 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:$1 User:node}
I180314 22:05:53.874671 148049 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:node}
I180314 22:05:53.874817 148303 storage/replica_command.go:851  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180314 22:05:53.880005 148391 storage/replica_command.go:851  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180314 22:05:53.884327 148467 storage/replica_command.go:851  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I180314 22:05:53.888349 148470 storage/replica_command.go:851  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I180314 22:05:53.889102 148049 sql/event_log.go:121  [n1] Event: "alter_table", target: 4, info: {TableName:system.public.users Statement:ALTER TABLE system.public.users ADD COLUMN IF NOT EXISTS "isRole" BOOL NOT NULL DEFAULT false User:node MutationID:1 CascadeDroppedViews:[]}
I180314 22:05:53.891460 148049 sql/lease.go:352  [n1] publish: descID=4 (users) version=2 mtime=2018-03-14 22:05:53.891372962 +0000 UTC
I180314 22:05:53.892353 148423 storage/replica_command.go:851  [split,n1,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I180314 22:05:53.896007 148049 sql/lease.go:352  [n1] publish: descID=4 (users) version=3 mtime=2018-03-14 22:05:53.895909366 +0000 UTC
I180314 22:05:53.896386 148317 storage/replica_command.go:851  [split,n1,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I180314 22:05:53.899677 148049 sql/backfill.go:134  [n1] Running backfill for "users", v=3, m=1
I180314 22:05:53.900845 148515 storage/replica_command.go:851  [split,n1,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180314 22:05:53.904987 148531 storage/replica_command.go:851  [split,n1,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
I180314 22:05:53.907698 148049 sql/lease.go:352  [n1] publish: descID=4 (users) version=4 mtime=2018-03-14 22:05:53.907577512 +0000 UTC
I180314 22:05:53.912137 148049 sql/event_log.go:121  [n1] Event: "finish_schema_change", target: 4, info: {MutationID:1}
I180314 22:05:53.913158 148049 sql/lease.go:275  publish (count leases): descID=4 name=users version=3 count=1
I180314 22:05:53.913732 148548 sql/lease.go:803  new lease: 4("users") ver=4:1521065499.833824350,0, refcount=0
I180314 22:05:53.950821 148049 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:gen_random_uuid()::STRING User:node}
I180314 22:05:53.951879 148049 server/server.go:1308  [n1] done ensuring all necessary migrations have run
I180314 22:05:53.951910 148049 server/server.go:1311  [n1] serving sql connections
I180314 22:05:53.953537 148049 storage/replica_command.go:851  [n1,s1,r5/1:/System/ts{d-e}] initiating a split of this range at key /System/tsd/test.metric/source1/10s/2017-12-14T22:00:00Z [r21]
I180314 22:05:53.954262 148417 sql/event_log.go:121  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:45607} Attrs: Locality: ServerVersion:2.0-1} ClusterID:b973e615-8683-4de6-a792-05f17a258c47 StartedAt:1521065153808222179 LastUp:1521065153808222179}
I180314 22:05:53.961412 148049 util/stop/stopper.go:473  quiescing; tasks left:
1      storage.intentResolver: processing intents
1      node.Node: batch
W180314 22:05:53.961588 148552 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: result is ambiguous (server shutdown)
--- FAIL: test/TestTimeSeriesMaintenanceQueueServer (1.790s)
ts_maintenance_queue_test.go:316: got datapoints [{1510697155000000000 100} {1515881155000000000 200}], expected [{1510697155000000000 100} {1515881155000000000 200} {1521065155000000000 300}], diff: [[]tspb.TimeSeriesDatapoint[2] != []tspb.TimeSeriesDatapoint[3]]
------- Stdout: -------
W180314 22:05:53.797250 148049 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180314 22:05:53.800763 148049 server/server.go:721  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180314 22:05:53.802697 148049 server/config.go:538  [n?] 1 storage engine initialized
I180314 22:05:53.802767 148049 server/config.go:541  [n?] RocksDB cache size: 128 MiB
I180314 22:05:53.802778 148049 server/config.go:541  [n?] store 0: in-memory, size 0 B
I180314 22:05:53.806893 148049 server/node.go:374  [n?] **** cluster b973e615-8683-4de6-a792-05f17a258c47 has been created
I180314 22:05:53.806928 148049 server/server.go:1110  [n?] **** add additional nodes by specifying --join=127.0.0.1:45607
I180314 22:05:53.807768 148049 storage/store.go:1394  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180314 22:05:53.807888 148049 server/node.go:504  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=4.9 KiB), ranges=1, leases=0, writes=0.00, bytesPerReplica={p10=5043.00 p25=5043.00 p50=5043.00 p75=5043.00 p90=5043.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00}
I180314 22:05:53.807925 148049 server/node.go:352  [n1] node ID 1 initialized
I180314 22:05:53.807979 148049 gossip/gossip.go:332  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:45607" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:1 > 
I180314 22:05:53.808067 148049 storage/stores.go:331  [n1] read 0 node addresses from persistent storage
I180314 22:05:53.808198 148049 server/node.go:645  [n1] connecting to gossip network to verify cluster ID...
I180314 22:05:53.808219 148049 server/node.go:670  [n1] node connected via gossip and verified as part of cluster "b973e615-8683-4de6-a792-05f17a258c47"
I180314 22:05:53.808235 148049 server/node.go:446  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180314 22:05:53.809176 148049 server/server.go:1228  [n1] starting https server at 127.0.0.1:37913
I180314 22:05:53.809204 148049 server/server.go:1229  [n1] starting grpc/postgres server at 127.0.0.1:45607
I180314 22:05:53.809214 148049 server/server.go:1230  [n1] advertising CockroachDB node at 127.0.0.1:45607
W180314 22:05:53.809234 148049 sql/jobs/registry.go:233  [n1] unable to get node liveness: node not in the liveness table
I180314 22:05:53.813344 147875 storage/replica_command.go:851  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I180314 22:05:53.817822 148107 storage/replica_command.go:851  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
I180314 22:05:53.822739 148289 storage/replica_command.go:851  [split,n1,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I180314 22:05:53.830576 148338 storage/replica_command.go:851  [split,n1,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
I180314 22:05:53.835926 147873 storage/replica_command.go:851  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
I180314 22:05:53.842976 148341 storage/replica_command.go:851  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I180314 22:05:53.848675 148354 storage/replica_command.go:851  [split,n1,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I180314 22:05:53.853805 148348 storage/replica_command.go:851  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I180314 22:05:53.858662 148350 storage/replica_command.go:851  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180314 22:05:53.862924 148049 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:node}
I180314 22:05:53.864208 148402 storage/replica_command.go:851  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180314 22:05:53.869645 148311 storage/replica_command.go:851  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180314 22:05:53.870429 148049 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:$1 User:node}
I180314 22:05:53.874671 148049 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:node}
I180314 22:05:53.874817 148303 storage/replica_command.go:851  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180314 22:05:53.880005 148391 storage/replica_command.go:851  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180314 22:05:53.884327 148467 storage/replica_command.go:851  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I180314 22:05:53.888349 148470 storage/replica_command.go:851  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I180314 22:05:53.889102 148049 sql/event_log.go:121  [n1] Event: "alter_table", target: 4, info: {TableName:system.public.users Statement:ALTER TABLE system.public.users ADD COLUMN IF NOT EXISTS "isRole" BOOL NOT NULL DEFAULT false User:node MutationID:1 CascadeDroppedViews:[]}
I180314 22:05:53.891460 148049 sql/lease.go:352  [n1] publish: descID=4 (users) version=2 mtime=2018-03-14 22:05:53.891372962 +0000 UTC
I180314 22:05:53.892353 148423 storage/replica_command.go:851  [split,n1,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I180314 22:05:53.896007 148049 sql/lease.go:352  [n1] publish: descID=4 (users) version=3 mtime=2018-03-14 22:05:53.895909366 +0000 UTC
I180314 22:05:53.896386 148317 storage/replica_command.go:851  [split,n1,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I180314 22:05:53.899677 148049 sql/backfill.go:134  [n1] Running backfill for "users", v=3, m=1
I180314 22:05:53.900845 148515 storage/replica_command.go:851  [split,n1,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180314 22:05:53.904987 148531 storage/replica_command.go:851  [split,n1,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
I180314 22:05:53.907698 148049 sql/lease.go:352  [n1] publish: descID=4 (users) version=4 mtime=2018-03-14 22:05:53.907577512 +0000 UTC
I180314 22:05:53.912137 148049 sql/event_log.go:121  [n1] Event: "finish_schema_change", target: 4, info: {MutationID:1}
I180314 22:05:53.913158 148049 sql/lease.go:275  publish (count leases): descID=4 name=users version=3 count=1
I180314 22:05:53.913732 148548 sql/lease.go:803  new lease: 4("users") ver=4:1521065499.833824350,0, refcount=0
I180314 22:05:53.950821 148049 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:gen_random_uuid()::STRING User:node}
I180314 22:05:53.951879 148049 server/server.go:1308  [n1] done ensuring all necessary migrations have run
I180314 22:05:53.951910 148049 server/server.go:1311  [n1] serving sql connections
I180314 22:05:53.953537 148049 storage/replica_command.go:851  [n1,s1,r5/1:/System/ts{d-e}] initiating a split of this range at key /System/tsd/test.metric/source1/10s/2017-12-14T22:00:00Z [r21]
I180314 22:05:53.954262 148417 sql/event_log.go:121  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:45607} Attrs: Locality: ServerVersion:2.0-1} ClusterID:b973e615-8683-4de6-a792-05f17a258c47 StartedAt:1521065153808222179 LastUp:1521065153808222179}
I180314 22:05:53.961412 148049 util/stop/stopper.go:473  quiescing; tasks left:
1      storage.intentResolver: processing intents
1      node.Node: batch
W180314 22:05:53.961588 148552 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: result is ambiguous (server shutdown)ts_maintenance_queue_test.go:316: got datapoints [{1510698125000000000 100} {1515882125000000000 200}], expected [{1510698125000000000 100} {1515882125000000000 200} {1521066125000000000 300}], diff: [[]tspb.TimeSeriesDatapoint[2] != []tspb.TimeSeriesDatapoint[3]]
------- Stdout: -------
W180314 22:21:58.992608 137882 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180314 22:21:59.016582 137882 server/server.go:721  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180314 22:21:59.020469 137882 server/config.go:538  [n?] 1 storage engine initialized
I180314 22:21:59.020600 137882 server/config.go:541  [n?] RocksDB cache size: 128 MiB
I180314 22:21:59.020635 137882 server/config.go:541  [n?] store 0: in-memory, size 0 B
I180314 22:21:59.062901 137882 server/node.go:374  [n?] **** cluster 19acc337-360e-44f2-aa50-9123e7740c3a has been created
I180314 22:21:59.063088 137882 server/server.go:1110  [n?] **** add additional nodes by specifying --join=127.0.0.1:34011
I180314 22:21:59.077829 137882 storage/store.go:1394  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180314 22:21:59.078966 137882 server/node.go:504  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=4.9 KiB), ranges=1, leases=0, writes=0.00, bytesPerReplica={p10=5043.00 p25=5043.00 p50=5043.00 p75=5043.00 p90=5043.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00}
I180314 22:21:59.079456 137882 server/node.go:352  [n1] node ID 1 initialized
I180314 22:21:59.079895 137882 gossip/gossip.go:332  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:34011" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:1 > 
I180314 22:21:59.082284 137882 storage/stores.go:331  [n1] read 0 node addresses from persistent storage
I180314 22:21:59.083728 137882 server/node.go:645  [n1] connecting to gossip network to verify cluster ID...
I180314 22:21:59.084046 137882 server/node.go:670  [n1] node connected via gossip and verified as part of cluster "19acc337-360e-44f2-aa50-9123e7740c3a"
I180314 22:21:59.084482 137882 server/node.go:446  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180314 22:21:59.086743 137882 server/server.go:1228  [n1] starting https server at 127.0.0.1:41085
I180314 22:21:59.086876 137882 server/server.go:1229  [n1] starting grpc/postgres server at 127.0.0.1:34011
I180314 22:21:59.086942 137882 server/server.go:1230  [n1] advertising CockroachDB node at 127.0.0.1:34011
W180314 22:21:59.087246 137882 sql/jobs/registry.go:233  [n1] unable to get node liveness: node not in the liveness table
I180314 22:21:59.109870 138119 storage/replica_command.go:851  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I180314 22:21:59.167683 137755 storage/replica_command.go:851  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
I180314 22:21:59.223493 137882 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:node}
I180314 22:21:59.317181 138123 storage/replica_command.go:851  [split,n1,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I180314 22:21:59.389111 137873 storage/replica_command.go:851  [split,n1,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
I180314 22:21:59.452073 138096 storage/replica_command.go:851  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
W180314 22:21:59.464560 138163 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "sql txn implicit" id=d34df7ed key=/Table/SystemConfigSpan/Start rw=true pri=0.01403173 iso=SERIALIZABLE stat=PENDING epo=0 ts=1521066119.362462836,0 orig=1521066119.362462836,0 max=1521066119.362462836,0 wto=false rop=false seq=14
I180314 22:21:59.479581 137882 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:$1 User:node}
I180314 22:21:59.517332 138125 storage/replica_command.go:851  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I180314 22:21:59.563482 137882 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:node}
I180314 22:21:59.587381 138179 storage/replica_command.go:851  [split,n1,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I180314 22:21:59.650342 138169 storage/replica_command.go:851  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I180314 22:21:59.707337 138172 storage/replica_command.go:851  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180314 22:21:59.761255 138212 storage/replica_command.go:851  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180314 22:21:59.809888 138184 storage/replica_command.go:851  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180314 22:21:59.821416 137882 sql/event_log.go:121  [n1] Event: "alter_table", target: 4, info: {TableName:system.public.users Statement:ALTER TABLE system.public.users ADD COLUMN IF NOT EXISTS "isRole" BOOL NOT NULL DEFAULT false User:node MutationID:1 CascadeDroppedViews:[]}
I180314 22:21:59.864610 137882 sql/lease.go:352  [n1] publish: descID=4 (users) version=2 mtime=2018-03-14 22:21:59.8631306 +0000 UTC
I180314 22:21:59.873162 138227 storage/replica_command.go:851  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180314 22:21:59.930378 138242 storage/replica_command.go:851  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180314 22:21:59.957192 137882 sql/lease.go:352  [n1] publish: descID=4 (users) version=3 mtime=2018-03-14 22:21:59.955466602 +0000 UTC
I180314 22:21:59.981232 138258 storage/replica_command.go:851  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I180314 22:22:00.017782 137882 sql/backfill.go:134  [n1] Running backfill for "users", v=3, m=1
I180314 22:22:00.036404 138234 storage/replica_command.go:851  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I180314 22:22:00.088387 138017 storage/replica_command.go:851  [split,n1,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I180314 22:22:00.136863 138278 storage/replica_command.go:851  [split,n1,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I180314 22:22:00.144746 137882 sql/lease.go:352  [n1] publish: descID=4 (users) version=4 mtime=2018-03-14 22:22:00.143634751 +0000 UTC
I180314 22:22:00.191113 138279 storage/replica_command.go:851  [split,n1,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180314 22:22:00.217736 137882 sql/event_log.go:121  [n1] Event: "finish_schema_change", target: 4, info: {MutationID:1}
I180314 22:22:00.240157 137882 sql/lease.go:275  publish (count leases): descID=4 name=users version=3 count=1
I180314 22:22:00.252094 138312 sql/lease.go:803  new lease: 4("users") ver=4:1521066480.691648094,0, refcount=0
I180314 22:22:00.259745 138315 storage/replica_command.go:851  [split,n1,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
I180314 22:22:00.479759 137882 sql/event_log.go:121  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:gen_random_uuid()::STRING User:node}
I180314 22:22:00.489283 137882 server/server.go:1308  [n1] done ensuring all necessary migrations have run
I180314 22:22:00.489533 137882 server/server.go:1311  [n1] serving sql connections
I180314 22:22:00.503725 137882 storage/replica_command.go:851  [n1,s1,r5/1:/System/ts{d-e}] initiating a split of this range at key /System/tsd/test.metric/source1/10s/2017-12-14T22:00:00Z [r21]
I180314 22:22:00.511433 138285 sql/event_log.go:121  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:34011} Attrs: Locality: ServerVersion:2.0-1} ClusterID:19acc337-360e-44f2-aa50-9123e7740c3a StartedAt:1521066119084129017 LastUp:1521066119084129017}
I180314 22:22:00.557073 137882 util/stop/stopper.go:473  quiescing; tasks left:
1      storage.intentResolver: processing intents
1      [async] transport racer
I180314 22:22:00.557181 138118 kv/transport_race.go:67  transport race promotion: ran 24 iterations on up to 792 requests
I180314 22:22:00.557782 137882 util/stop/stopper.go:473  quiescing; tasks left:
1      storage.intentResolver: processing intents
W180314 22:22:00.557464 138359 storage/intent_resolver.go:388  [n1,s1] failed to cleanup transaction intents: could not GC completed transaction anchored at /Local/Range/System/tsd/RangeDescriptor: node unavailable; try another peer

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/ts_maintenance_queue_test.go
+++ b/pkg/storage/ts_maintenance_queue_test.go
@@ -237,7 +237,10 @@ func TestTimeSeriesMaintenanceQueueServer(t *testing.T) {
 	// periods; this simplifies verification.
 	seriesName := "test.metric"
 	sourceName := "source1"
-	now := tsrv.Clock().PhysicalNow()
+	// "now" is five minutes in the past to avoid any sort of shenanigans with the
+	// various adjustments we make in the very-recent-past to create consistent
+	// graphs.
+	now := tsrv.Clock().PhysicalNow() - int64(5*time.Minute)
 	nearPast := now - (tsdb.PruneThreshold(ts.Resolution10s) * 2)
 	farPast := now - (tsdb.PruneThreshold(ts.Resolution10s) * 4)
 	sampleDuration := ts.Resolution10s.SampleDuration()

